conf/systemd/packetfence-httpd.admin_dispatcher.service

```diff
@@ -1,12 +1,9 @@
 # Copyright (C) Inverse inc.
 [Unit]
 Description=PacketFence HTTP Admin Dispatcher
-Wants=packetfence-base.target packetfence-config.service packetfence-iptables.service
-After=packetfence-base.target packetfence-config.service packetfence-iptables.service
+Wants=packetfence-base.target packetfence-config.service
+After=packetfence-base.target packetfence-config.service
 Before=packetfence-haproxy-admin.service
-Before=packetfence-docker-iptables.service
-Requires=packetfence-docker-iptables.service
-PartOf=packetfence-docker-iptables.service
```

Comment:
Why there is no call to generate the firewalld config ?

Reply:
Because it is starting at the installation, and at this point pfconfig is empty.
There is no config done.
iptables is not started either at the same point docker-iptables service is a fake one. https://github.com/inverse-inc/packetfence/blob/devel/conf/systemd/packetfence-docker-iptables.service#L9

Reply:
In fact that iptables service dependency at boot is using the fact that the service iptables during pf installation will fail.
Thanks to the failure, at the exiit of iptables, new rules are defined.
```
-- Journal begins at Wed 2024-07-10 11:19:44 UTC, ends at Wed 2024-07-10 11:44:36 UTC. --                                                                                                          [13/1674]
Jul 10 11:29:53 172-234-151-109 systemd[1]: Started PacketFence Iptables configuration.
Jul 10 11:29:55 172-234-151-109 packetfence[30662]: -e(30662) ERROR: unable to connect to database: Access denied for user 'pf'@'localhost' (pf::db::db_connect)
Jul 10 11:29:56 172-234-151-109 packetfence[30662]: -e(30662) ERROR: unable to connect to database: Access denied for user 'pf'@'localhost' (pf::db::db_connect)
Jul 10 11:29:57 172-234-151-109 packetfence[30662]: -e(30662) ERROR: unable to connect to database: Access denied for user 'pf'@'localhost' (pf::db::db_connect)
Jul 10 11:29:58 172-234-151-109 packetfence[30662]: -e(30662) ERROR: unable to connect to database: Access denied for user 'pf'@'localhost' (pf::db::db_connect)
Jul 10 11:29:59 172-234-151-109 packetfence[30662]: -e(30662) ERROR: unable to connect to database: Access denied for user 'pf'@'localhost' (pf::db::db_connect)
Jul 10 11:30:00 172-234-151-109 packetfence[30662]: -e(30662) ERROR: unable to connect to database: Access denied for user 'pf'@'localhost' (pf::db::db_connect)
Jul 10 11:30:01 172-234-151-109 packetfence[30662]: -e(30662) ERROR: unable to connect to database: Access denied for user 'pf'@'localhost' (pf::db::db_connect)
Jul 10 11:30:02 172-234-151-109 systemd[1]: Stopping PacketFence Iptables configuration...
Jul 10 11:30:02 172-234-151-109 packetfence[30662]: -e(30662) ERROR: unable to connect to database: Access denied for user 'pf'@'localhost' (pf::db::db_connect)
Jul 10 11:30:03 172-234-151-109 packetfence[30662]: -e(30662) ERROR: unable to connect to database: Access denied for user 'pf'@'localhost' (pf::db::db_connect)
Jul 10 11:30:04 172-234-151-109 packetfence[30662]: -e(30662) ERROR: unable to connect to database: Access denied for user 'pf'@'localhost' (pf::db::db_connect)
Jul 10 11:30:05 172-234-151-109 sudo[31331]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -F
Jul 10 11:30:05 172-234-151-109 sudo[31331]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31331]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31334]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -X
Jul 10 11:30:05 172-234-151-109 sudo[31334]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31334]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31337]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -t nat -F
Jul 10 11:30:05 172-234-151-109 sudo[31337]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31337]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31340]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -t nat -X
Jul 10 11:30:05 172-234-151-109 sudo[31340]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31340]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31343]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -t mangle -F
Jul 10 11:30:05 172-234-151-109 sudo[31343]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31343]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31346]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -t mangle -X
Jul 10 11:30:05 172-234-151-109 sudo[31346]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31346]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31349]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -P INPUT ACCEPT
Jul 10 11:30:05 172-234-151-109 sudo[31349]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31349]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31352]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -P FORWARD ACCEPT
Jul 10 11:30:05 172-234-151-109 sudo[31352]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31352]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31355]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -P OUTPUT ACCEPT
Jul 10 11:30:05 172-234-151-109 sudo[31355]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31355]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31358]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -t nat -N DOCKER
Jul 10 11:30:05 172-234-151-109 sudo[31358]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31358]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31361]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -t nat -A PREROUTING -m addrtype --dst-type LOCAL -j DOCKER
Jul 10 11:30:05 172-234-151-109 sudo[31361]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31361]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31364]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -t nat -A OUTPUT ! -d 127.0.0.0/8 -m addrtype --dst-type LOCAL -j DOCKER
Jul 10 11:30:05 172-234-151-109 sudo[31364]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31364]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31367]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -t nat -A POSTROUTING -s 100.64.0.0/10 ! -o docker0 -j MASQUERADE
Jul 10 11:30:05 172-234-151-109 sudo[31367]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31367]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31370]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -t nat -A DOCKER -i docker0 -j RETURN
Jul 10 11:30:05 172-234-151-109 sudo[31370]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31370]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 systemd[1]: packetfence-iptables.service: Succeeded.
Jul 10 11:30:05 172-234-151-109 systemd[1]: Stopped PacketFence Iptables configuration.
Jul 10 11:30:05 172-234-151-109 systemd[1]: packetfence-iptables.service: Consumed 5.222s CPU time.
```
So it is tricky, because we are using iptables in order to have the rules defined when it is failing.

Reply:
In fact that iptables service dependency at boot is using the fact that the service iptables during pf installation will fail.
Thanks to the failure, at the exiit of iptables, new rules are defined.
```
-- Journal begins at Wed 2024-07-10 11:19:44 UTC, ends at Wed 2024-07-10 11:44:36 UTC. --                                                                                                          [13/1674]
Jul 10 11:29:53 172-234-151-109 systemd[1]: Started PacketFence Iptables configuration.
Jul 10 11:29:55 172-234-151-109 packetfence[30662]: -e(30662) ERROR: unable to connect to database: Access denied for user 'pf'@'localhost' (pf::db::db_connect)
Jul 10 11:29:56 172-234-151-109 packetfence[30662]: -e(30662) ERROR: unable to connect to database: Access denied for user 'pf'@'localhost' (pf::db::db_connect)
Jul 10 11:29:57 172-234-151-109 packetfence[30662]: -e(30662) ERROR: unable to connect to database: Access denied for user 'pf'@'localhost' (pf::db::db_connect)
Jul 10 11:29:58 172-234-151-109 packetfence[30662]: -e(30662) ERROR: unable to connect to database: Access denied for user 'pf'@'localhost' (pf::db::db_connect)
Jul 10 11:29:59 172-234-151-109 packetfence[30662]: -e(30662) ERROR: unable to connect to database: Access denied for user 'pf'@'localhost' (pf::db::db_connect)
Jul 10 11:30:00 172-234-151-109 packetfence[30662]: -e(30662) ERROR: unable to connect to database: Access denied for user 'pf'@'localhost' (pf::db::db_connect)
Jul 10 11:30:01 172-234-151-109 packetfence[30662]: -e(30662) ERROR: unable to connect to database: Access denied for user 'pf'@'localhost' (pf::db::db_connect)
Jul 10 11:30:02 172-234-151-109 systemd[1]: Stopping PacketFence Iptables configuration...
Jul 10 11:30:02 172-234-151-109 packetfence[30662]: -e(30662) ERROR: unable to connect to database: Access denied for user 'pf'@'localhost' (pf::db::db_connect)
Jul 10 11:30:03 172-234-151-109 packetfence[30662]: -e(30662) ERROR: unable to connect to database: Access denied for user 'pf'@'localhost' (pf::db::db_connect)
Jul 10 11:30:04 172-234-151-109 packetfence[30662]: -e(30662) ERROR: unable to connect to database: Access denied for user 'pf'@'localhost' (pf::db::db_connect)
Jul 10 11:30:05 172-234-151-109 sudo[31331]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -F
Jul 10 11:30:05 172-234-151-109 sudo[31331]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31331]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31334]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -X
Jul 10 11:30:05 172-234-151-109 sudo[31334]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31334]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31337]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -t nat -F
Jul 10 11:30:05 172-234-151-109 sudo[31337]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31337]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31340]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -t nat -X
Jul 10 11:30:05 172-234-151-109 sudo[31340]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31340]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31343]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -t mangle -F
Jul 10 11:30:05 172-234-151-109 sudo[31343]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31343]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31346]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -t mangle -X
Jul 10 11:30:05 172-234-151-109 sudo[31346]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31346]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31349]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -P INPUT ACCEPT
Jul 10 11:30:05 172-234-151-109 sudo[31349]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31349]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31352]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -P FORWARD ACCEPT
Jul 10 11:30:05 172-234-151-109 sudo[31352]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31352]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31355]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -P OUTPUT ACCEPT
Jul 10 11:30:05 172-234-151-109 sudo[31355]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31355]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31358]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -t nat -N DOCKER
Jul 10 11:30:05 172-234-151-109 sudo[31358]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31358]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31361]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -t nat -A PREROUTING -m addrtype --dst-type LOCAL -j DOCKER
Jul 10 11:30:05 172-234-151-109 sudo[31361]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31361]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31364]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -t nat -A OUTPUT ! -d 127.0.0.0/8 -m addrtype --dst-type LOCAL -j DOCKER
Jul 10 11:30:05 172-234-151-109 sudo[31364]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31364]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31367]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -t nat -A POSTROUTING -s 100.64.0.0/10 ! -o docker0 -j MASQUERADE
Jul 10 11:30:05 172-234-151-109 sudo[31367]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31367]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 sudo[31370]:     root : PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -t nat -A DOCKER -i docker0 -j RETURN
Jul 10 11:30:05 172-234-151-109 sudo[31370]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=0)
Jul 10 11:30:05 172-234-151-109 sudo[31370]: pam_unix(sudo:session): session closed for user root
Jul 10 11:30:05 172-234-151-109 systemd[1]: packetfence-iptables.service: Succeeded.
Jul 10 11:30:05 172-234-151-109 systemd[1]: Stopped PacketFence Iptables configuration.
Jul 10 11:30:05 172-234-151-109 systemd[1]: packetfence-iptables.service: Consumed 5.222s CPU time.
```
So it is tricky, because we are using iptables in order to have the rules defined when it is failing.